gui-lib/mred/private/wxme/editor-canvas.rkt

```diff
@@ -139,6 +139,44 @@
 
 ;; ----------------------------------------
 
+;; This is not a subclass of timer% so that the events can be scheduled
+;; as precisely as possible. timer%s schedule the next nofication
+;; `interval' milliseconds _after_ the notification has been handled,
+;; whereas this class subtracts the duration of `notify' from the
+;; interval.
+(define scroll-timer%
+  (class object%
+    (init-field canvas)
+    (super-new)
+
+    (define interval (exact->inexact (* 1/60 1000)))
+    (define running #f)
+
+    (define (loop)
+      (when running
+        (define deadline
+          (alarm-evt (+ (current-inexact-milliseconds) interval)))
+        (when canvas
+          (if (send canvas is-shown-to-root?)
+              (send canvas on-scroll-flush)
+              (stop)))
+        (yield deadline)
+        (loop)))
```

Comment:
A loop that `yield`s does not seem like a good idea, because it inverts control in a way that is less composable. For example, imagine with two such loops; the second one cannot return and continue until the inner one ends. I guess the idea here is that the loops would be limited by input events. Still, I think using a `timer%` or a looping thread with `queue-callback` (which should differ only in terms of priority) is the way to go.

Reply:
Thanks! That makes sense. I've reverted back to the `timer%` implementation.